frontend/src/api/prometheus/__tests__/useQueryRangeResourceData.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { TimeframeTitle } from '~/pages/modelServing/screens/types';
+import useQueryRangeResourceData from '~/api/prometheus/useQueryRangeResourceData';
+import { TimeframeStep, TimeframeTimeRange } from '~/pages/modelServing/screens/const';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import * as usePrometheusQueryRangeModule from '~/api/prometheus/usePrometheusQueryRange';
+import { mockPrometheusQueryResponse } from '~/__mocks__/mockPrometheusQueryResponse';
+
+describe('useQueryRangeResourceData', () => {
+  const active = true;
+  const query = 'testQuery';
+  const end = 123456;
+  const timeframe: TimeframeTitle = TimeframeTitle.ONE_HOUR;
+  const responsePredicate = jest.fn();
+  const namespace = 'testNamespace';
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should call usePrometheusQueryRange with correct arguments and restructure the returned data', async () => {
+    const spy = jest.spyOn(usePrometheusQueryRangeModule, 'default');
+    const mockedResponse = { data: { result: mockPrometheusQueryResponse({}) } };
+    const restructuredData = [
+      mockedResponse.data.result.data,
+      true,
+      undefined,
+      expect.any(Function),
+      false,
+    ];
+
+    spy.mockReturnValue(restructuredData as never);
```

Comment:
Do not cast to `never`. Create an appropriate return value that matches the return type so that future changes in types that would otherwise fail the test don't get missed.